lib/internal/assert/calltracker.js

```diff
@@ -0,0 +1,100 @@
+'use strict';
+
+const {
+  Error,
+  SafeSet,
+} = primordials;
+
+const {
+  codes: {
+    ERR_UNAVAILABLE_DURING_EXIT,
+  },
+} = require('internal/errors');
+const AssertionError = require('internal/assert/assertion_error');
+const { validateUint32 } = require('internal/validators');
+
+const noop = () => {};
+
+class CallTracker {
+
+  #callChecks = new SafeSet()
+
+  constructor({ onExit = false } = {}) {
+    if (onExit === true) {
+      process.on('exit', () => this.verify());
+    }
+  }
```

Comment:
The constructor needs to have some input type validation.. The test should also include checks on the input types.

```suggestion
const {
  validateBoolean,
  validateObject,
  validateUint32,
} = require('internal/validators');

const noop = () => {};

class CallTracker {

  #callChecks = new SafeSet()

  constructor(options = { onExit: true }) {
    validateObject(options, 'options');
    const { onExit } = options;
    validateBoolean(onExit, 'options.onExit');
    if (onExit)
      process.on('exit', () => this.verify());
  }
```